### ISSUE TITLE
Long initialization of modules

### ISSUE THREAD

**Issue by Orvacko** [jsreport/jsreport#191]
Hi, i was trying to init jsreport in node more effective, but i cant find way. Init of modules takes sometimes 20+ seconds. 
- log
```
2016-03-21T15:08:52.260Z - info: Export wrapper module - init - start
2016-03-21T15:09:05.859Z - info: Export wrapper module - init - finished
```
- config
```
    var jsreport = require('jsreport')({
        rootDirectory: config.root + '../../node_modules/',
        tempDirectory: config.sapExportDirs.root,
        dataDirectory: config.sapExportDirs.root,
        extensions: [
            "handlebars",
            "templates",
            "wkhtmltopdf",
            "text"
        ],
        logger: { providerName: "console" },
        wkhtmltopdf: {
            allowLocalFilesAccess: true
        }
    });

    jsreport.init().then(function(){

        deferred.resolve();

    }).catch(function(error) {
        deferred.reject(error);
    });
```

Do you have some tip how t omake faster local init, is tehre any option to disable looking for modules when i have explicit say, which modules to use? (it looks like it is long because it looks in every folder in node_modules)

Thx

**Comment by pofider** [comment]
Hi. Yes, we will have to do something with this. If you have many other modules and slow disk, this can take some time.

To speed it up now you can explicitly `use` extensions you like. This will avoid crawling the node_modules. 

```js
var jsreport = require('./')({rootDirectory:__dirname...})
jsreport.use(require('jsreport-express')())
jsreport.use(require('jsreport-templates')())
jsreport.use(require('jsreport-handlebars')())

jsreport.init()
```

The docs for this are [here](https://github.com/jsreport/jsreport-core#extensions-auto-discovery)

**Comment by Orvacko** [comment]
Thank you for info and help. I have 1 problem with this - npm module jsreport-text, it throws error (because this module doesnt have index.js in package):
```
Error: Cannot find module ...
```
- full config
```
jsreport.use(require(config.root + '../../node_modules/jsreport-handlebars')());
jsreport.use(require(config.root + '../../node_modules/jsreport/node_modules/jsreport-templates')());
jsreport.use(require(config.root + '../../node_modules/jsreport-wkhtmltopdf')());
jsreport.use(require(config.root + '../../node_modules/jsreport-text')());
```

Do you have some workaround here ? (or i must do some?)

Thank you

**Comment by pofider** [comment]
`jsreport-text` had no support for `use`. I fixed this now and published `jsreport-test@0.2.0`. Please install that one, I will add it also to the next jsreport package hotfix.

**Comment by Orvacko** [comment]
Thank you :)

**Comment by Orvacko** [comment]
Hm, i have small issue, i cant find your package, npm install fails.
```
$ npm search jsreport-test
npm WARN Building the local index for the first time, please be patient
NAME          DESCRIPTION                                        AUTHOR   DATE       VERSION KEYWORDS
jsreport-test jsreport recipe for rendering plain text from html =pofider 2015-11-13 0.0.1   jsreport
```

**Comment by pofider** [comment]
There was a typo, I updated the last comment 30 seconds ago:),

The package name is `jsreport-text`

**Comment by Orvacko** [comment]
Perfect ! Thank you very much. Time goes down from 15-20 sec to 1-2 sec. #

**Comment by pofider** [comment]
Great!

I think we should improve this behavior by caching the locations of the extensions. The crawling would be done then only if the directory was modified since the last crawl.

Or maybe adding a requirement that extension needs to have always `jsreport.config.js` in it's root directory would speed the crawling enough.

We will definitely think about it. Thank you.

This problem will be solved in #192
